core/server/worker/src/main/java/alluxio/worker/grpc/GrpcExecutors.java

```diff
@@ -42,25 +41,22 @@
           THREAD_STOP_MS, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(512),
           ThreadFactoryUtils.build("AsyncCacheManagerExecutor-%d", true)));
 
-  public static final ExecutorService BLOCK_READER_EXECUTOR =
-      new ImpersonateThreadPoolExecutor(new ThreadPoolExecutor(THREADS_MIN,
-          ServerConfiguration.getInt(PropertyKey.WORKER_NETWORK_BLOCK_READER_THREADS_MAX),
-          THREAD_STOP_MS, TimeUnit.MILLISECONDS, new SynchronousQueue<>(),
-          ThreadFactoryUtils.build("BlockDataReaderExecutor-%d", true)));
+  public static final ExecutorService BLOCK_IO_EXECUTOR =
+      new ImpersonateThreadPoolExecutor(new ThreadPoolExecutor(
+          ServerConfiguration.getInt(PropertyKey.WORKER_NETWORK_BLOCK_IO_THREADS),
+          ServerConfiguration.getInt(PropertyKey.WORKER_NETWORK_BLOCK_IO_THREADS),
+          THREAD_STOP_MS, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(
```

Comment:
Now that the queue has a very high bound and low thread count, shouldn't we just use `SynchronousQueue`?
Otherwise blocking the request at queue while the queue has 3K+ items may not be the best thing.

Reply:
+ AFAIR, `LinkedBlockingQueue` will not cause `RejectedExecution`

Reply:
LinkedBlockingQueue will cause a RejectedExecution exception if you limit its capacity. here's an example that will throw one:

```java
@Test
  public void testRejectedExecutionException() throws Exception {
    ExecutorService svc = new ThreadPoolExecutor(1, 1, 1, TimeUnit.MINUTES,
        new LinkedBlockingQueue<>(1));
    svc.submit(() -> LockSupport.parkNanos(10000000));
    svc.submit(() -> System.out.print("Hello, "));
    svc.submit(() -> System.out.print("World"));
  }
```

Reply:
`SynchronousQueue<>` will not allow **any** queued tasks to the executor which limits our ability to receive incoming requests.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/SynchronousQueue.html

> A blocking queue in which each insert operation must wait for a corresponding remove operation by another thread, and vice versa. A synchronous queue does not have any internal capacity, not even a capacity of one.

Reply:
Right. Got it.